concourse/tasks/build-and-upload.yml

```diff
@@ -8,9 +8,11 @@ inputs:
   - name: tools
   - name: repo
   - name: version
-  - name: m2
 outputs:
   - name: out
+caches:
```

Comment:
There’s no real reason that these need to be hidden directories.  In fact, it’s probably easier that they aren’t, so that when you `fly hijack` it’s obvious that they’re there and being used.  I’d just call them `gradle` and `maven` and then link them to their hidden locations.

Reply:
Yep, makes sense.